hpx/runtime/threads/policies/lockfree_queue_backends.hpp

```diff
@@ -21,11 +26,17 @@ namespace hpx { namespace threads { namespace policies
 struct lockfree_fifo;
 struct lockfree_lifo;
 
+///////////////////////////////////////////////////////////////////////////////
 // FIFO
 template <typename T>
 struct lockfree_fifo_backend
 {
+#if defined(HPX_HAVE_CXX11_STD_ATOMIC_128BIT)
     typedef boost::lockfree::deque<T> container_type;
+#else
+    typedef boost::lockfree::queue<T> container_type;
```

Comment:
By the way, this could actually always be `queue` instead of `deque`. I simply changed all of them to `deque` in #3172 because I thought it doesn't hurt to have them all use the same underlying datastructure. The other queues actually need `deque`.

Reply:
As is turns out, this change or the change to `lockfree::stack` above has caused a performance regression. I will prepare a PR reverting this part of the patch.

Reply:
See #4111 for the patch that reverts this particular change.